features/polyExplorer/src/components/entityList/entityList.jsx

```diff
@@ -58,19 +17,26 @@ function getStartGroups(entityGroups) {
     return keys.pop();
 }
 
-const EntityList = () => {
-    const { entities, globalData, activeFilters, handleRemoveFilter } =
-        useContext(ExplorerContext);
-    const onRemoveFilter = handleRemoveFilter;
-    const filteredEntities = activeFilters.apply(entities);
-    const entityGroups = groupEntities(filteredEntities);
+export default function EntityList({ entityGroups }) {
```

Comment:
NIT: It would be more react-ey to have it as `export default EntityList({ EntityGroups }) => {`

Reply:
Why? React does not like functions?

Reply:
Stateless functional components (non-class components and how one should use react nowadays) are conventionally done by using arrow functions. I forgot the const thats why it was misleading (since now it is a function :D)
The function keyword is usually used with logical functions and it could be misleading this way

Reply:
According to [this](https://betterprogramming.pub/difference-between-regular-functions-and-arrow-functions-f65639aba256) the difference is respect to the use of `this` as well as the access to `.arguments`. Does this have any influence here? (And sorry, this is a noob question, I am perfectly fine with what you say)

Reply:
Totally fine, never needing an excuse to ask questions should be company policy :D! 
As you said in highly this dependent code as Object methods this makes a huge difference, here I would say none at all except for dev understanding

Reply:
Hm, not a big fan of cargo culting this one - I only see two relevant differences for the arrow function in this particular case:

1. It binds `this` from the outer scope (not using `this`, but we definitely wouldn't want that behaviour because in a component, we cannot tell what it's bound to)
2. It's _more_ to type.

My personal rule for when to use arrow functions is:

1. When it's shorter (e.g. when the only statement in the block would be a `return`)
2. When I explicitly want to access `this` from the outer scope

Now, rule (1) applies when we do something like this:

```
export default () => {
    ...
}
```

Which I actually do in the code I just added :D But this:

```
const Foo = () => {
...
};
export default Foo;
```

IMHO doesn't quite beat:

```
function Foo() {
...
}
export default Foo;
```

That said, we currently don't have any explicit rules for when to use arrow functions and when not, so it's up to the taste of the author unless it's inconsistent with the surrounding code.

If there's some reasoning for why the React community does this ("to not confuse functional components with other functions" isn't good enough IMHO :P), I wouldn't mind changing it.

Reply:
As a side note on this -> I now added some _preprocessing_ logic I'm kinda proud of: Basically a function that processes the input (props), normalises it and sets sane defaults. The one thing it currently does is to set the `sideLoad` attribute if it wasn't specified by the caller, based on the length of the group keys. The next thing I was planning to use it for is to support an _array_ of entities rather than an object. In that function, I would simply transform it into an object with only a single group, and then add some logic to the component to not render labels if there's only one group.

Given how _functions returning a component_ are a pretty powerful compositioning instrument for the kind of UI library we want to build (shape shifting based on the input data), I kinda even _like_ the idea of not distinguishing between _component functions_ and other functions - it's easier (also for beginners, I think) to design stuff like that when they don't attempt to create any distinctions.